osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/smartmetering/ClockStatus.java

```diff
@@ -51,6 +51,9 @@ public boolean hasStatusBits() {
         return this.statusBits != null;
     }
 
+    // this is a getter, meaning it should return what the actual value is. Since it is possible that the actual
+    // value is set to null, it should also return this in that case.
+    @SuppressWarnings("squid:S1168")
```

Comment:
I don't think we should suppress this warning, and return an empty set.
Write unit tests for all the code calling this method to make sure everything keeps on working ok.